documentation/projects/proposals/rekognition_data/20240510-implementation_plan_machine_generated_tags_frontend.md

```diff
@@ -0,0 +1,199 @@
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### Create a wrapper `VMediaTags` component
+
+This component will handle filtering the tags into two sets: machine-generated
+and source-provided. It will also display the headings and the link to the
+"About tags" page. A new child component, `VCollapsibleTagSection`, will handle
+displaying of the tags in several rows, collapsing them if there are more than
+three rows of tags.
+
+Currently, we deduplicate all tags to only display unique tags. When we display
+the tags by generation type, there will probably be duplicates between the
+source and the generated tags. However, each section individually will still be
+normalized to prevent duplicates.
+
+The generated tags section should split the tags by different providers into
+subsections, and should use the provider of the tag as the section label. For
+instance, if there are tags by both `recognition` and `clarifai`, there will be
+two sub-sections in the Generated tags section: tags generated by `recognition`
+and tags generated by `clarifai`. In the future, if there are more providers,
+there could be more tag sections. The tags in each subsection will be ordered by
+their accuracy.
+
+There is no need to create a separate component for machine-generated tags, the
+slot will enable us to display the additional accuracy information.
+
+### Add "About tags" description page
+
+We should add a page to explain the users the difference between where the
+Openverse tags come from, the difference between the source and generated tags,
+and a short description of the machine-generated tag providers. This page should
+be linked from the Generated tags section.
+
+## Dependencies
+
+This project needs the API to add the `unstable__provider` property to tags in
+media responses, which was implemented in
+[Expose provider in the API tags response #4280](https://github.com/WordPress/openverse/pull/4280)
+
+### Feature flags
+
+<!-- List feature flags/environment variables that will be utilised in the development of this plan. -->
+
+The project will contain at most 2 PRs that can be reverted, so the feature
+flags are not necessary.
+
+### Infrastructure
+
+<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->
+
+No infrastructure changes will be necessary, we will only update the single
+result pages of the Nuxt frontend.
+
+## Alternatives
+
+<!-- Describe any alternatives considered and why they were not chosen or recommended. -->
+
+The tags will be displayed in two sections: "Source tags" and "Generated tags".
+
+For the machine-generated tags, we want to add a link to the page that describes
```

Comment:
It looks like this is missing a bit of information at the end.

Reply:
Thank you for pointing this out. This was actually extra that should have been removed.